client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -124,13 +123,11 @@ public ControllerImpl(final URI controllerURI, final ControllerImplConfig config
      *
      * @param channelBuilder The channel builder to connect to the service instance.
      * @param config         The configuration for this client implementation.
-     * @param executor       The executor service to be used internally.
      */
     @VisibleForTesting
-    public ControllerImpl(ManagedChannelBuilder<?> channelBuilder, final ControllerImplConfig config,
-                          final ScheduledExecutorService executor) {
+    public ControllerImpl(ManagedChannelBuilder<?> channelBuilder, final ControllerImplConfig config) {
         Preconditions.checkNotNull(channelBuilder, "channelBuilder");
-        this.executor = executor;
+        this.executor = Executors.newScheduledThreadPool(config.getThreadpoolSize());
```

Comment:
use `ExecutorServiceHelpers.newScheduledThreadPool`. That will do the same, configure the thread pool properly, and give it a name so we can identify its threads in the logs.

Reply:
Also, you MUST close this executor when you close this class. Otherwise your application will hang around forever. 
This class needs a `close()` method (I don't see one), properly make use of the `closed` variable (it is currently not set anywhere from what I see), and have its `close()` method invoked by whoever is owning it.

Reply:
done